tests/acceptance/00_basics/def.json/def_preferred.json/def_preferred.cf

```diff
@@ -0,0 +1,42 @@
+body common control
+{
+      # Feel free to avoid including "default.cf.sub" and define your
+      # own bundlesequence for simple tests
+
+      inputs => { "../../../default.cf.sub" };
+      bundlesequence  => { default("$(this.promise_filename)") };
+}
+
+#######################################################
+
+bundle agent init
+# Initialize the environment and prepare for test
+{
+
+}
+
+bundle agent test
+# Activate policy for behaviour you wish to inspect
+{
+  meta:
+    "description" -> { "CFE-3656" }
+      string => "def_preferred.json is preferred over def.json if it exists.";
+  classes:
+    "ok"
+      if => and(
+        strcmp(length(variablesmatching(".*\.good_var")), "1"),
+        strcmp(length(variablesmatching(".*\.bad_var")),  "0"),
+        fileexists("$(this.promise_dirname)/def.json"),
+        fileexists("$(this.promise_dirname)/def_preferred.json")),
+      scope => "namespace";
+}
+
+bundle agent check
+# Check the result of the test
+{
+  reports:
+    ok::
```

Comment:
I would use a dcs method for this instead of just pass/fail but nipicks (like no need to have an empty agent init bundle) the test seems reasonable.

I probably would have tested for exactly def.good_var and def.bad_var instead of using variablesmatching which  would find those vars if they ever get introduced to the dcs library (via deffault.cf.sub).